### ISSUE TITLE
Add support for DynamoDB transactions

### ISSUE THREAD

**Issue by jesucarr** [aws/aws-sdk-js#1004]
The AWS-SDK for Java has [support for transactions](https://aws.amazon.com/blogs/aws/dynamodb-transaction-library/) since 2013. Time to add it to JavaScript as well!

**Comment by LiuJoyceC** [comment]
Thanks for the suggestion! We will look into this feature request for the JavaScript SDK.

**Comment by Tamal** [comment]
Can you also add transaction support in Golang?

**Comment by chrisradek** [comment]
@Tamal
For requests to the Go version of the SDK, please create an issue here:
https://github.com/aws/aws-sdk-go

**Comment by anujanavare** [comment]
hello Joyce, can we expect the transaction support in the javascript sdk anytime soon?

**Comment by lexakozakov** [comment]
it's good the have.

**Comment by QAnders** [comment]
Any update on this? 
It is a must have to be able to handle user work flows...

**Comment by funwun** [comment]
Any update on this?

**Comment by efatatrisatya** [comment]
+1

**Comment by fsarrate** [comment]
+1

**Comment by issfire** [comment]
+1

**Comment by sherifnada** [comment]
+1

**Comment by Florentijn** [comment]
+1

**Comment by gozup** [comment]
+1

**Comment by sesha-nyllabs** [comment]
+1

**Comment by farzadpanahi** [comment]
+1

**Comment by guotai** [comment]
+1

**Comment by Hmachalani** [comment]
+1

**Comment by FerencH** [comment]
+1

**Comment by Manelmc** [comment]
+1

**Comment by melaniaandrisan** [comment]
+1

**Comment by artyomtkachenko** [comment]
+1

**Comment by Marchhill** [comment]
+1

**Comment by sherifnada** [comment]
@LiuJoyceC is there any update on prioritizing this issue? It would make life a lot easier for us :)

**Comment by samikarak** [comment]
+1

**Comment by chrisradek** [comment]
Thank you for all the requests for this feature. We appreciate the feedback and are watching this thread. I don't have any updates yet, but will report here as soon as I do.

**Comment by tanzeelrana** [comment]
+1
:+1:

**Comment by nolde** [comment]
+1

**Comment by adisembiring** [comment]
👍

**Comment by keego** [comment]
+1 pretty pretty please

**Comment by urbien** [comment]
we need it to write to two DynamoDB tables. Alternative is have one Lambda call another, but this creates unneeded latency and ads costs.

**Comment by tanzeelrana** [comment]
One solution I found was using meta data for items, lock the items related to the transactional queries and once all the queries are complete release the locks

**Comment by udag** [comment]
+1

**Comment by jbehmer** [comment]
+1

**Comment by bitbrained** [comment]
+1

**Comment by NqCompiler** [comment]
+1

**Comment by evandroguedes** [comment]
+1

**Comment by UmeshDotNet** [comment]
+1

**Comment by superandrew213** [comment]
+1

**Comment by candidasa** [comment]
+1

**Comment by kuashe** [comment]
This issue has been open for more than a year now , can we get an update maybe ?

**Comment by voidale** [comment]
+1

**Comment by aryehischechter** [comment]
+1

**Comment by trevorhreed** [comment]
tl;dr Use a thumbs-up reaction instead of commenting `+1`/`me too`/`etc.` to show support for a Github ticket.

For those who may not be aware, the best way to show your support for a ticket on Github is to use the reactions feature. It is much easier for a project owner to query for popular tickets when that popularity is expressed through reactions. Commenting with `+1` is less effective and sends an email to everyone subscribed to the ticket, thus motivating them to unsubscribe so they stop getting as many emails; when people unsubscribe from a ticket, the ticket gets less attention, less support, less help. So, in a sense, whenever you comment `+1`, you're actually decreasing support for that ticket. Also, it's much harder to find relevant information about a task/bug (for example, possible workarounds or updates from the project owners) when you have to scroll through a lot of `+1` comments. Finally, every time you comment `+1` on a ticket a puppy dies. So, if for no other reason, do it for the puppies; their lives are on the line:

![maxresdefault](https://user-images.githubusercontent.com/3343902/30872588-d340055e-a2a7-11e7-91af-e969a58eb8ed.jpg)

**Comment by bluszcz** [comment]
@trevorhreed +1, well said

**Comment by samikarak** [comment]
I guess that all of us would appreciate to just get an update on this, just to be informed when/if you are tackling this !?!

**Comment by jodevak** [comment]
+1

**Comment by rubemz** [comment]
+1

**Comment by chrisradek** [comment]
Hey everyone, first off, thank you for your feedback.

We've discussed this internally and have decided not to include this feature directly as part of the AWS JavaScript SDK. We have forwarded all your feedback to the DynamoDB team, and ask that you also request support for this on their [forums](https://forums.aws.amazon.com/forum.jspa?forumID=131). If you create a feature request there I will happily forward that to them as well.

**Comment by mciccarello** [comment]
Why the hell not?

**Comment by evandroguedes** [comment]
You guys are just ignoring a huge userbase of Amazon AWS Lambda users with this decision. That's not make any sense!

**Comment by alowsarwar** [comment]
When can we expect good transaction management for JS SDK, or any other workaround avl?

**Comment by fredvollmer** [comment]
While I respect the decision of the team to focus on other things, an explanation of why this was passed over internally would be nice to have. Is there a technical reason why this feature is less important in the JS SDK than in the Java SDK? Is there a different approach to atomic transactions that we as developers should be using?

It would be great for the DynamoDB team to include this support directly, but in the meantime, I just fail to see why the JS SDK should not strive for feature parity with the Java SDK (in this particular respect.)

**Comment by fredvollmer** [comment]
Also, I haven't looked extensively at how this is implemented in Java, but would you be interested in a PR?

**Comment by qm3ster** [comment]
@tanzeelrana Your approach sounds like that's what's happening with the `7N + 4` write units the official transaction api requires.

**Comment by srchase** [comment]
Hello All!

DynamoDB Transactions are now available as of version [2.365.0](https://github.com/aws/aws-sdk-js/blob/61f89137708f199bec5dfe7bf8f4754719324501/CHANGELOG.md#23650).

**Comment by sherifnada** [comment]
@srchase +1
